goreq.go

```diff
@@ -436,7 +437,9 @@ func (r Request) NewRequest() (*http.Request, error) {
 	}
 
 	var bodyReader io.Reader
-	if b != nil && r.Compression != nil {
+	if r.BodyReader != nil {
```

Comment:
Why is this necessary?. If `Body` is a reader then the `prepareRequestBody` will return a `Reader` which will be assigned here: https://github.com/franela/goreq/pull/126/files#diff-04563bc79ed43224b4d9ebd57a2ad279R456